vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -176,9 +294,174 @@ void MediaTwitter::SavePublisherInfo(
                           duration,
                           window_id,
                           callback);
+
   if (!media_key.empty()) {
     ledger_->SetMediaPublisherInfo(media_key, publisher_key);
   }
 }
 
+void MediaTwitter::OnSaveMediaVisit(
+    ledger::Result result,
+    ledger::PublisherInfoPtr info) {
+}
+
+void MediaTwitter::FetchDataFromUrl(
+    const std::string& url,
+    braveledger_media::FetchDataFromUrlCallback callback) {
+  ledger_->LoadURL(url,
+                   std::vector<std::string>(),
+                   std::string(),
+                   std::string(),
+                   ledger::URL_METHOD::GET,
+                   callback);
+}
+
+void MediaTwitter::OnMediaActivityError(const ledger::VisitData& visit_data,
+                                        uint64_t window_id) {
+  std::string url = TWITTER_TLD;
+  std::string name = TWITTER_MEDIA_TYPE;
+
+  DCHECK(!url.empty());
+
+  ledger::VisitData new_data;
+  new_data.domain = url;
+  new_data.url = "https://" + url;
+  new_data.path = "/";
+  new_data.name = name;
+
+  ledger_->GetPublisherActivityFromUrl(window_id, new_data, std::string());
+}
+
+void MediaTwitter::ProcessActivityFromUrl(
+    uint64_t window_id,
+    const ledger::VisitData& visit_data) {
+  // not all url's are publisher specific
+  if (IsExcludedPath(visit_data.path)) {
+    OnMediaActivityError(visit_data, window_id);
+    return;
+  }
+
+  const std::string user_name = GetUserNameFromUrl(visit_data.path);
+  const std::string media_key = GetMediaKey(user_name);
+
+  if (!media_key.empty()) {
```

Comment:
Would be better to fail early here, i.e. `if (media_key.empty() { return
 }`

Reply:
done